src/test.js

```diff
@@ -329,21 +332,26 @@ Test.prototype = {
 
 			// Source of Test
 			source: this.stack
-		} );
-
-		if ( module.testsRun === numberOfTests( module ) ) {
-			logSuiteEnd( module );
-
-			// Check if the parent modules, iteratively, are done. If that the case,
-			// we emit the `suiteEnd` event and trigger `moduleDone` callback.
-			let parent = module.parentModule;
-			while ( parent && parent.testsRun === numberOfTests( parent ) ) {
-				logSuiteEnd( parent );
-				parent = parent.parentModule;
+		} ).then( function() {
+			if ( module.testsRun === numberOfTests( module ) ) {
+				return logSuiteEnd( module );
 			}
-		}
-
-		config.current = undefined;
+		} ).then( function() {
```

Comment:
I'm a bit confused:

1. Why are there multiple `.then()` calls in sequence?
1. There appear to be two areas of logic where `logSuiteEnd()` is called. Why do we have two areas of logic and why are we separating the handling for this module vs any parent modules?

Reply:
Sorry for the delay @platinumazure.
1. Since each `.then()` handles an individual promise, which from the previous non promise based implementation, seemed to be running in sequential order. Hence I chained them together. 
2. I was going with the old logic, and didn't put too much thought into this. Do you have any suggestions?

Reply:
I had some time to revisit that code, and made changes to both 1 and 2. 
Particularly for 2, I changed it so that all modules (this module or parent's modules) are handled in the same code path